src/pyvcam_driver.py

```diff
@@ -0,0 +1,537 @@
+        set or only a single ROI is supported, this function will over-write that ROI. Otherwise,
+        this function will attempt to append this ROI to the list.
+
+        :param s1: Serial coordinate 1.
+        :param p1: Paralled coordinate 1.
+        :width: Num pixels in serial direction.
+        :height: Num pixels in parallel direction.
+
+        :return: None.
+        """
+        self.cam.set_roi(s1, p1, width, height)
+    
+    def shape(self, roi_index: int=0) -> tuple[int, int]:
+        """
+        Returns the reshape factor to be used when acquiring a ROI. This is equivalent to an acquired images shape.
+        
+        :param roi_index: Index of ROI
+
+        :return:
+        """
+        return self.cam.shape(roi_index)
+    
+    def read_enum(self, param_id: int) -> dict[str, int]:
+        """
+        Returns all settings names paired with their values of a parameter.
+
+        :param param_id: The parameter ID.
+
+        :return: A dictionary containing strings mapped to values.
+        """
+        return self.cam.read_enum(param_id)
+    
+    def binning(self) -> tuple[int, int]:
+        """
+        Returns current serial and parallel binning values
+
+        :return:
+        """
+        return self.cam.binning
+    
+    def set_binning(self, value: Union[tuple[int, int], int]) -> None:
+        """
+        Changes binning. A single value wills et a square binning.
+        Binning cannot be changed directly on the camera; but is used for setting up 
+        acquisitions and returning correctly shaped images returned from get_frame and get_live_frame.
+        Binning settings for individual ROIs is not supported.
+
+        :param value: Desired binning value as a tuple, or int for a square binning.
+
+        :return: None.
+        """
+        self.cam.binning = value
+    
+    def chip_name(self) -> str:
+        """
+        Returns the camera sensor's name as a string.
+
+        :return:
+        """
+        return self.cam.chip_name
+    
+    def post_trigger_delay(self) -> int:
+        """
+        Returns the last acquisition's post-trigger delay reported by the camera in ms.
+        
+        :return:
+        """
+        return self.cam.post_trigger_delay
+    
+    def pre_trigger_delay(self) -> int:
+        """
+        Returns the last acquisition's pre-trigger delay reported by the camera in ms.
+
+        :return:
+        """
+        return self.cam.pre_trigger_delay
+    
+    def scan_line_time(self) -> int:
+        """
+        Returns the scan line time in ns.
+
+        :return:
+        """
+        return self.cam.scan_line_time
+    
+    def sensor_size(self) -> tuple[int, int]:
+        """
+        Returns the sensor size of the current camera in a tuple 
+        in the form (serial sensor size, parallel sensor size)
+
+        :return:
+        """
+        return self.cam.sensor_size
+    
+    def speed_table_index(self) -> int:
```

Comment:
```suggestion
    def get_speed_table_index(self) -> int:
```
For symmetry with `set_speed_table_index`.